buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/trialsinterface.py

```diff
@@ -0,0 +1,111 @@
+from hdmf.backends.hdf5.h5_utils import H5DataIO
+
+
+def access_behavior_property_safe(property, parent, behavior_mat):
+    trial_info = behavior_mat["behavior"]["trials"]
+    nest_depth = len(trial_info["position_trcat"])
+
+    value = parent[property]
+    if nest_depth > 1:
+        value = [num for sublist in value for num in sublist]  # Flatten the list if large depth
+
+    assert len(value) == len(
+        trial_info["recordings"]
+    )  # Save access properties should have the same length as the number of recordings
+
+    return value
+
+
+def isNaN(num):
+    return num != num
+
+
+def to_direction(arr):
+    output = []
+    for value in arr:
+        if isNaN(value):
+            output.append("timeout")
+
+        elif value == 0:
+            output.append("right")
+
+        else:
+            output.append("left")
+
+    return output
+
+
+# Add trial table from the behavior file
+class HuszarTrialsInterface(BaseDataInterface):
+    def __init__(self, folder_path: FolderPathType):
+        super().__init__(folder_path=folder_path)
+
+    def run_conversion(self, nwbfile: NWBFile, metadata: dict, stub_test: bool = False):
+        self.session_path = Path(self.source_data["folder_path"])
+        self.session_id = self.session_path.stem
+
+        # Add trial table from the behavior file
+        behavior_file_path = self.session_path / f"{self.session_id}.Behavior.mat"
+        behavior_mat = loadmat_scipy(behavior_file_path, simplify_cells=True)
+
+        trial_info = behavior_mat["behavior"]["trials"]
+        trial_interval_list = access_behavior_property_safe("trial_ints", trial_info, behavior_mat)
+
+        familiar_final_idx = len(trial_info["trial_ints"])
+        if len(trial_info["position_trcat"]) > 1:
+            familiar_final_idx = len(trial_info["trial_ints"][0])
+
+        data = []
+        is_familiar_maze = []
+
+        for idx, times in enumerate(trial_interval_list):
+            data.append(
+                dict(
+                    start_time=float(times[0]),
+                    stop_time=float(times[1]),
+                )
+            )
+
+            is_familiar_maze.append(idx < familiar_final_idx)
+
+        [nwbfile.add_trial(**row) for row in sorted(data, key=lambda x: x["start_time"])]
+
+        visited_arm_data = to_direction(trial_info["visitedArm"])
+        expected_arm_data = to_direction(trial_info["expectedArm"])
+
+        nwbfile.add_trial_column(
+            name="visited_matched_expected",
+            description="A boolean representing whether the expected and visited arm of the trial matches",
+            data=trial_info["choice"],
+        )
+
+        nwbfile.add_trial_column(
+            name="visited_arm",
+            description="An integer value representing the visited arm of the trial",
+            data=H5DataIO(visited_arm_data, compression="gzip"),
+        )
+
+        nwbfile.add_trial_column(
+            name="expected_arm",
```

Comment:
I notice now that it is only the `expected_arm` that seems to have `NaN`s; I had proposed the `"timeout"` interpretation before, but can you shed light on if that makes any sense from the paper or other description of experiment setup? What exactly is the 'expected' movement and how is it determined?

Reply:
Ah you are right. 

## The First Maze
Since this is an alternation task, it looks like the animal initializes what the start condition is based on its first choice. After this initialization of the first "expected" arm with the first visited arm, the next expected column is the reverse of the previous one.

We could keep NaN or add a "none" string.

## The Second (unfamiliar) Maze
At least for the uploaded file, this is a linear maze. It seems that there is no expected arm here—which makes the reward structure of the experiment somewhat opaque at this point. 

Either option for handling NaN seems like it would also be appropriate here.

Reply:
"none" does sound much better in this case; thanks for digging into it!